hub/core/tensor.py

```diff
@@ -200,6 +256,77 @@ def _check_array_and_tensor_are_compatible(tensor_meta: dict, array: np.ndarray)
         )
 
 
+def read(image_path: str, check_meta: bool = True):
```

Comment:
in #924 i implemented a function `hub.load`. please copy this file and it's exact folder location https://github.com/activeloopai/Hub/blob/feature/2.0/hub-auto/hub/util/load.py 

and implement your code in there

Reply:
i can resolve conflicts later but this function is very ambiguous and probably doesn't belong in `hub.core.tensor`

Reply:
also when you do this, could you please also break up this massive function into some smaller methods?